src/main/resources/Listings/Listings.drl

```diff
@@ -126,3 +117,36 @@ rule "Session is listed less or equal 50 percent 7 days before start"
                                   new ProblemReference($s)));
 end
 
+
+rule "The case type on the listing request does not match a case type associated to the session type of the session"
+    when
+        Year($year: value)
+        Month($month: value)
+        Day($day: value)
+        $session : Session(DateTimeUtils.isAfter(start, $year, $month, $day))
+        $hearingPart : HearingPart(sessionId == $session.id)
+        $sessionType : SessionType(code == $session.sessionType && !this.containsCaseTypeByCode($hearingPart.caseType))
+    then
+        insertLogical(new Problem(ProblemTypes.Listing_request_types_violation,
+                                  ProblemSeverities.Warn,
+                                  "The case type on the listing request does not match a case type associated to the session type of the session",
+                                  new ProblemReference($hearingPart),
+                                  new ProblemReference($session)));
+end
+
+rule "The hearing type on the listing request does not match a hearing type associated to the session type of the session"
```

Comment:
Just wondering if we could have just one rule, because in the current state of the application we do not allow user to pick e.g. hearing type which is not associated to case type. If my understanding is correct with these 2 rules we would generate 2 problems when someone chooses incorrect case type and then hearing type, because he cannot pick the right one. Not sure if that's the right behaviour?